src/bin/bench-tps.rs

```diff
@@ -444,63 +438,71 @@ fn main() {
                 .help("number of threads"),
         )
         .arg(
-            Arg::with_name("seconds")
-                .short("s")
-                .long("seconds")
-                .value_name("NUM")
+            Arg::with_name("duration")
+                .long("duration")
+                .value_name("SECS")
                 .takes_value(true)
-                .help("send transactions for this many seconds"),
+                .help("run benchmark for SECS seconds then exit, default is forever"),
         )
         .arg(
             Arg::with_name("converge-only")
-                .short("c")
                 .long("converge-only")
                 .help("exit immediately after converging"),
         )
         .arg(
             Arg::with_name("sustained")
                 .long("sustained")
-                .help("Use sustained performance mode vs. peak mode. This overlaps the tx generation with transfers."),
+                .help("use sustained performance mode vs. peak mode. This overlaps the tx generation with transfers."),
         )
         .arg(
             Arg::with_name("tx_count")
                 .long("tx_count")
-                .value_name("NUMBER")
+                .value_name("NUM")
                 .takes_value(true)
-                .help("number of transactions to send in a single batch")
+                .help("number of transactions to send per batch")
         )
         .get_matches();
 
-    let network = matches
-        .value_of("network")
-        .unwrap()
-        .parse()
-        .unwrap_or_else(|e| {
+    let network = if let Some(addr) = matches.value_of("network") {
+        addr.parse().unwrap_or_else(|e| {
             eprintln!("failed to parse network: {}", e);
             exit(1)
-        });
-
-    let id = read_keypair(matches.value_of("keypair").unwrap()).expect("client keypair");
-
-    if let Some(t) = matches.value_of("threads") {
-        threads = t.to_string().parse().expect("integer");
-    }
-
-    if let Some(n) = matches.value_of("num_nodes") {
-        num_nodes = n.to_string().parse().expect("integer");
-    }
-
-    if let Some(s) = matches.value_of("seconds") {
-        time_sec = s.to_string().parse().expect("integer");
-    }
-
-    if let Some(s) = matches.value_of("tx_count") {
-        tx_count = s.to_string().parse().expect("integer");
-    }
-
-    if matches.is_present("sustained") {
-        sustained = true;
-    }
+        })
+    } else {
+        socketaddr!("127.0.0.1:8001")
+    };
+
+    let id = if let Some(k) = matches.value_of("keypair") {
+        read_keypair(k).expect("can't read client keypair")
+    } else {
+        Keypair::new()
```

Comment:
I think we may want to keep the same keypair for each run. This would create a new one right? Otherwise we can leak tokens. Although if it's running forever, then it should be ok.

Reply:
yes: this creates a new, ephemeral key if no key is passed on the command line.  The command-line invoker was generating a new keypair on each new deployment of the client.

I'd like to better understand "leak tokens".  Aren't they air dropped, essentially unlimited whereever this client runs?

Reply:
Well if your client stops and then you restart with a new key you will have left some tokens in all the random accounts that the client. Do that enough times and you run the mint out of tokens? I never saw this happen myself but @mvines put some logic in to try and prevent that so I assume it may be a problem...

If the client is not restarting though then it's not really a problem.